src/View/SSViewer_Scope.php

```diff
@@ -359,27 +343,90 @@ public function next()
             return false;
         }
 
-        return $this->itemIterator->key();
+        return true;
     }
 
     /**
-     * @param string $name
-     * @param array $arguments
-     * @return mixed
+     * Get the value that will be directly rendered in the template.
      */
-    public function __call($name, $arguments)
+    public function getOutputValue(string $name, array $arguments, string $type): string
     {
-        $on = $this->getCurrentItem();
-        if ($on instanceof ViewableData && $name === 'XML_val') {
-            $retval = $on->XML_val(...$arguments);
+        $retval = $this->getObj($name, $arguments, $type);
+        $this->resetLocalScope();
+        return $retval === null ? '' : $retval->__toString();
+    }
+
+    /**
+     * Get the value to pass as an argument to a method.
+     */
+    public function getValueAsArgument(string $name, array $arguments, string $type): mixed
```

Comment:
This new method controls how values are passed into function arguments from a template. It used to use `XML_val` and therefore template variables couldn't be passed.
Instead, the raw value is now passed.